### PR TITLE
#309: use custom thread-poll size

### PR DESCRIPTION
**Pull request by lautarobock**
#309: use custom thread-poll size to be sure than always have more threads available than requests in tests

### CODE REVIEW

**Comment by davvd** [comment]
@lautarobock Thanks, let me find someone who can review this pull request

**Comment by davvd** [comment]
@caarlos0 review this one, please

**Comment by caarlos0** [comment]
@rultor merge

File: src/test/java/org/takes/http/BkParallelTest.java [hunk 1]
```diff
@@ -80,12 +80,16 @@ public Response act(final Request req)
...
+            // @checkstyle AnonInnerLengthCheck (23 lines)
             new Runnable() {
                 @Override
                 public void run() {
                     try {
                         new FtBasic(
-                            new BkParallel(new BkBasic(take)),
+                            new BkParallel(
+                                new BkBasic(take),
+                                BkParallelTest.size(count)
```
Comment (yegor256) [inline src/test/java/org/takes/http/BkParallelTest.java]: @lautarobock why can't we just use `count`?

**Comment by yegor256** [comment]
@lautarobock I didn't get the logic... why not just `count`?

**Comment by lautarobock** [comment]
@yegor256 I want thread-poll greater than `Runtime.getRuntime().availableProcessors()` but thinking about this, if `Runtime.getRuntime().availableProcessors()` is lesser than count, It wont work. Let me fix it

**Comment by lautarobock** [comment]
@yegor256 Now is using `count`. But count has a max depending of threads available. I think this could be the cause of instability.

**Comment by yegor256** [comment]
@lautarobock I don't understand it... now we have 3 threads. you're making _less_ than 3. how this will improve the situation?

File: src/test/java/org/takes/http/BkParallelTest.java [hunk 2]
```diff
@@ -80,12 +83,16 @@ public Response act(final Request req)
...
+            // @checkstyle AnonInnerLengthCheck (23 lines)
             new Runnable() {
                 @Override
                 public void run() {
                     try {
                         new FtBasic(
-                            new BkParallel(new BkBasic(take)),
+                            new BkParallel(
+                                new BkBasic(take),
+                                count
```
Comment (lautarobock) [inline src/test/java/org/takes/http/BkParallelTest.java]: @yegor256 The idea is of using the same amount of threads for `BkParallel` and for requests. Previously used `availableProcessors()` and could be not enough.
And min between 3 and availableProcessors (lines 65..68) is to be sure that it have enough threads to make all request and can respond it.

Comment (yegor256) [inline src/test/java/org/takes/http/BkParallelTest.java]: @lautarobock but you will always have 1 or 2 or 3 threads, not more. now we always have 3. how do 1 or 2 solve the problem?

Comment (lautarobock) [inline src/test/java/org/takes/http/BkParallelTest.java]: @yegor256 in some cases there is not 3 available threads and I think that
the problem. Two threads are blocked and there isn't any more to count down
till 0 and release the others

Comment (lautarobock) [inline src/test/java/org/takes/http/BkParallelTest.java]: @yegor256 so? Do you think is right or Will I try with other solution?

Comment (yegor256) [inline src/test/java/org/takes/http/BkParallelTest.java]: @lautarobock so to solve the problem we need **more** threads, not less. Your code makes **less** threads than three.

**Comment by lautarobock** [comment]
@yegor256 now is using always same `count` value

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@elenavolokhova review this ticket please, for compliance with our [quality rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy pls

**Comment by elenavolokhova** [comment]
@caarlos0 
According to our [quality rules](http://www.teamed.io/qa.html):
 > The code reviewer found at least three problems in the code.
 
 There were no issues found during review.
Please confirm that you will try to find at least three problems next time or ask to reassign CR to another reviewer.

**Comment by caarlos0** [comment]
@elenavolokhova sure, I'll try

**Comment by elenavolokhova** [comment]
@davvd Quality is acceptable here.

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@elenavolokhova thank you, let's try to make it "good" next time

**Comment by davvd** [comment]
@caarlos0 just added 10 mins to @elenavolokhova (for QA), payment ID is `58631309`

Much appreciated! **17 mins** was added to your account, payment ID `AP-80R14472F47302116`, time spent is 119 hours and 19 mins

extra minutes for review comments (c=2)

added +17 to your rating, now it is equal to [+2049](http://www.netbout.com/b/35095?open=rating)

**Pull request merged**